scripts/tags_and_languages.py

```diff
@@ -0,0 +1,338 @@
+    ON products_unknown_tags.{tag_type}_tag = taxonomy_tags.lc_tag_name
+    WHERE taxonomy_tags.lc_tag_name IS NOT NULL
+    AND taxonomy_tags.lc != 'xx'
+    AND products_unknown_tags.{tag_type}_lc_and_tag != taxonomy_tags.lc_tag_name_concat
+    ''').df()
+    # remark:
+    #   AND products_unknown_tags.{tag_type}_lc_and_tag != taxonomy_tags.lc_tag_name_concat -> to prevent fr:angleterre and fr:angleterre
+
+    print(f"prepare log table - time {datetime.now()-time_start}")
+    conn_db = duckdb.connect(database='tags_and_languages.db', read_only=False)
+    conn_db.execute(f'''
+    DROP TABLE IF EXISTS products_to_update_{tag_type}
+    ''')
+    conn_db.execute(f'''
+    CREATE TABLE products_to_update_{tag_type} (
+        code TEXT,
+        lang TEXT,
+        old_tags_text TEXT,
+        new_tags_text TEXT,
+        updated_tags_text BOOLEAN,
+        updated BOOLEAN
+    )
+    ''')
+    for _, row in tags_to_update_df.iterrows():
+        code = row[0]
+        lang = row[1]
+        
+        existing_row = conn_db.execute(f"SELECT * FROM products_to_update_{tag_type} WHERE code = '{code}'").fetchone()
+        
+        if existing_row:
+            # Update the existing row
+            # existing_row[2] is tags as text already updated
+            # row[3] is unknown tag with lc
+            # row[4] is unknown tag
+            # row[8] is row[4] tag found in another language
+            new_tags_text, updated_tags_text = update_tags(existing_row[3], row[3], row[4], row[8])
+
+            conn_db.execute(f'''
+            UPDATE products_to_update_{tag_type}
+            SET new_tags_text = ?, updated_tags_text = CASE 
+                                    WHEN updated_tags_text = TRUE THEN TRUE 
+                                    ELSE ? 
+                                END
+            WHERE code = ?
+            ''', [new_tags_text, updated_tags_text, code])
+        else:
+            # Insert a new row
+            # row[2] is tags as text
+            # row[3] is unknown tag with lc
+            # row[4] is unknown tag
+            # row[8] is row[4] tag found in another language
+            new_tags_text, updated_tags_text = update_tags(row[2], row[3], row[4], row[8])
+
+            conn_db.execute(f'''
+            INSERT INTO products_to_update_{tag_type} (code, lang, old_tags_text, new_tags_text, updated_tags_text, updated)
+            VALUES (?, ?, ?, ?, ?, ?)
+            ''', [code, lang, row[2], new_tags_text, updated_tags_text, False])
+
+    print(f"finally update products - time {datetime.now()-time_start}")
+    all_rows = conn_db.execute(f"SELECT * products_to_update_{tag_type}").fetchall()
+
+    for row in all_rows:
+        code = row[0]
+        lang = row[1]
+
+        try:
+            country = mapping_languages_countries[lang]
+        except KeyError:
+            print(f"ERROR: language {lang} is not referenced in mapping_languages_countries")
+            sys.exit()
```

Comment:
do you need a sys.exit() ? isn't an error log enough ?

Reply:
the code interrupt here
the user will have to manually update mapping_languages_countries
and resume the job